### ISSUE TITLE
[BUG] Composite charts should not mix plugins together

### ISSUE THREAD

**Issue by ktsaou** [netdata/netdata-cloud#351]
When we have multiple different operating systems, like `linux` and `freebsd`, the `context` of charts remain the same, but the dimensions are different for each operating system.

This is not a problem for single node dashboards, because every single node dashboard only one version of them.

However, in composite charts, this becomes messy.

Examples:

Page faults chart from both linux and freebsd:
Dimensions `cow` and `memory` come from freebsd. Dimensions `minor` and `major` come from linux.

![image](https://user-images.githubusercontent.com/2662304/158649190-4f2c9afa-9c89-45af-840e-36d8c2ebeed5.png)

Memory chart from both linux and freebsd:
FreeBSD has `free`, `active`, `inactive`, `wired`, `laundry` and `buffers`
Linux has `free`, `used`, `cached` and `buffers`

![image](https://user-images.githubusercontent.com/2662304/158649656-1faa71de-5fa2-4a4a-8e75-e8f2c0060055.png)

---

## Discussion

This issue is also related to https://github.com/netdata/netdata-cloud/issues/347

So, when we have a `context` that has been generated by 2 different plugins, the charts should not be merged in composite dashboards.

The fact that charts on composite dashboards have different aggregations (some charts come from 1 node, others come from 2 nodes, etc) makes the composite dashboard more difficult to understand. Just paying attention to the nodes or charts counter, is not the right way to present this information.

Something is missing here to give clarity to the dashboard.

This ticket is about discussing the right solution. No action to be taken for the moment.

**Comment by cakrit** [comment]
This is another that is probably for the agent, not the cloud, but we'll discuss first.

**Comment by hugovalente-pm** [comment]
without the work on being completed the only _quick and dirty_ option I see https://github.com/netdata/netdata-cloud/issues/354 is to tweek with families to have subsections for each

where we could have a have
* system.freebsd
* system.linux
* system.macos

what do you think @ktsaou @amalkov @papazach @novykh @stelfrag @ilyam8 ?

```json
		"mem.pgfaults": {
			"chartLabels": {},
			"chartType": "line",
			"context": "mem.pgfaults",
			"family": "system",
			"firstEntry": 1654507069,
			"lastEntry": 1654810642,
			"module": "/proc/vmstat",
			"plugin": "proc.plugin",
			"priority": 1040,
			"title": "Memory Page Faults",
			"units": "faults/s",
			"updateEvery": 1
		},
```

**Comment by hugovalente-pm** [comment]
just ping to see if you have any feedback on this @ktsaou @amalkov @papazach @novykh @stelfrag @ilyam8 ?

**Comment by novykh** [comment]
2-3 months later, I agree with the family grouping.

**Comment by hugovalente-pm** [comment]
for this one we can see what we will be doing on this one as well https://github.com/netdata/netdata-cloud/issues/550, I know that one was more describing PostegreSQL but we will need to have a rule(s) that could also help on this one

**Comment by amalkov** [comment]
@hugovalente-pm agree with you, this can be resolved together with #550

**Comment by hugovalente-pm** [comment]
I think this bug can be closed, this was raised for discussion and we identified that we have a short-term solution described here with https://github.com/netdata/netdata-cloud/issues/550 and the long-term solution is the  https://github.com/netdata/netdata-cloud/issues/354
